### PR TITLE
DOPE-264: correct the type of some String Functions

### PR DESCRIPTION
**Pull request by jansigi**

### CODE REVIEW

**Comment by jansigi** [comment]
@CodiumAI-Agent /review

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/MBPosition1Expression.kt [hunk 1]
```diff
@@ -19,3 +20,5 @@ fun mbPosition1(inStr: String, searchStr: TypeExpression<StringType>) =
 
 fun mbPosition1(inStr: String, searchStr: String) =
     mbPosition1(inStr.toDopeType(), searchStr.toDopeType())
+
+// test
```
[the remark this answers is not in the document]
Comment (jansigi) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/MBPosition1Expression.kt]: done

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/Position1Expression.kt [hunk 2]
```diff
@@ -3,10 +3,11 @@ package ch.ergon.dope.resolvable.expression.unaliased.type.function.stringfuncti
 import ch.ergon.dope.resolvable.expression.TypeExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.function.FunctionExpression
 import ch.ergon.dope.resolvable.expression.unaliased.type.toDopeType
+import ch.ergon.dope.validtype.NumberType
 import ch.ergon.dope.validtype.StringType
 
 class Position1Expression(inStr: TypeExpression<StringType>, searchStr: TypeExpression<StringType>) :
-    FunctionExpression<StringType>("POSITION1", inStr, searchStr)
+    FunctionExpression<NumberType>("POSITION1", inStr, searchStr)
```
Comment (pgruntz) [inline core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/Position1Expression.kt:10]: Why aren't any tests affected? Shouldn't you add some then?

**Review by pgruntz — APPROVED** [review]

File: core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/Position1Expression.kt (hunk 2 above)
Comment (jansigi) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/Position1Expression.kt:10]: The problem with the generic is that you can't realy test it exept if you write something like this:
`val actual: Position1Expression<NumberType> = position1("test", "test")` and we don't do that anywhere.
should i add that (i would need to add it to every function call that has uses a ValidType)?

Comment (pgruntz) [reply core/src/main/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/Position1Expression.kt:10]: Well I would expect a test like:
`...where position1("lel","lol") + 4 < 7`
This ensures that it is a number-type
(I guess this bug came up, becuase such a test was missing)

**Review by pgruntz — APPROVED** [review]

File: core/src/test/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/ReplaceExpressionTest.kt [hunk 3]
```diff
@@ -0,0 +1,241 @@
...
+        val replace = someString("replace")
+        val numberOfInstances = someInt()
+        val expected = ReplaceExpression(inStr, searchStr.toDopeType(), replace.toDopeType(), numberOfInstances.toDopeType())
+
+        val actual = replace(inStr, searchStr, replace, numberOfInstances)
+
+        assertEquals(expected.toDopeQuery(manager), actual.toDopeQuery(manager))
+    }
+
+    // asödlkjfölaskdj fölkasjd fölkasj dfölkasj döflkajsödlfj aösldjf
```
[the remark this answers is not in the document]
Comment (jansigi) [reply core/src/test/kotlin/ch/ergon/dope/resolvable/expression/unaliased/type/function/stringfunction/ReplaceExpressionTest.kt]: done

**Pull request merged**